### PR TITLE
RDoc-1823-subscription-exception copied files 4.0=>5.1

### PR DESCRIPTION
**Pull request by morhilai**

### CODE REVIEW

File: Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+Subscriptions are consumed by processing batches of documents received from the server. 
+A `SubscriptionWorker` object manages the documents processing and the communication between the client and the server according to a set of configurations received upon it's creation. 
+We've introduced several ways to create and configure a SubscriptionWorker, starting from just giving a subscription name, and ending with a detailed configuration object - `SubscriptionWorkerOptions`.
+
+In this page:
+
+[SubscriptionWorker lifecycle](../../../client-api/data-subscriptions/consumption/how-to-consume-data-subscription#subscriptionworker-lifecycle)  
+[Error handling](../../../client-api/data-subscriptions/consumption/how-to-consume-data-subscription#error-handling)  
+[Workers interplay](../../../client-api/data-subscriptions/consumption/how-to-consume-data-subscription#workers-interplay)
```
Comment (garayx) [inline Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown:15]:
```suggestion
[Workers interplay](../../../client-api/data-subscriptions/consumption/how-to-consume-data-subscription#worker-interplay)
```

File: Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  * The existing connection has a strategy that is `TakeOver`. In this case, the incoming connection will throw a SubscriptionInUseException exception.  
+{PANEL/}
+
+## Related Articles
+
+**Data Subscriptions**:
+
+- [What are Data Subscriptions](../../../client-api/data-subscriptions/what-are-data-subscriptions)
+- [How to Create a Data Subscription](../../../client-api/data-subscriptions/creation/how-to-create-data-subscription)
+- [How to Consume a Data Subscription](../../../client-api/data-subscriptions/consumption/how-to-consume-data-subscription)
```
Comment (garayx) [inline Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown:103]: remove this one, points to the current article

File: Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+{INFO/}
+
+{INFO: Reconnecting}
+In the cases above, we described situations in which a worker will try to reconnect with the server. There are two key `SubscriptionWorkerOptions` fields controlling this state:
+
+*  `TimeToWaitBeforeConnectionRetry` - The time that the worker will 'sleep' before trying to reconnect.
+* `MaxErroneousPeriod` - The maximum time in which the worker is allowed to be in erroneous state. After that time passes, the worker will stop trying to reconnect
+{INFO/}
+
+{INFO: `OnUnexpectedSubscriptionError`}
```
Comment (garayx) [inline Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown:73]:
```suggestion
{INFO: OnUnexpectedSubscriptionError}
```

File: Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+When this occurs, the worker will automatically try to reconnect again. This event is 
+useful for logging these unexpected exceptions.
+{INFO/}
+
+{PANEL/}
+
+{PANEL: Worker interplay}
+There can only be one active subscription worker working on a subscription. 
+Nevertheless, there are scenarios where it is required to interact between an existing subscription worker and one that tries to connect. 
+This relationship and interoperation is configured by the `SubscriptionConnectionOptions` `Strategy` field.
```
Comment (garayx) [inline Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown:85]:
```suggestion
This relationship and interoperation is configured by the `SubscriptionWorkerOptions` `Strategy` field.  
```

File: Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+useful for logging these unexpected exceptions.
+{INFO/}
+
+{PANEL/}
+
+{PANEL: Worker interplay}
+There can only be one active subscription worker working on a subscription. 
+Nevertheless, there are scenarios where it is required to interact between an existing subscription worker and one that tries to connect. 
+This relationship and interoperation is configured by the `SubscriptionConnectionOptions` `Strategy` field.  
+The strategy field is an enum, having the following values:
```
Comment (garayx) [inline Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown:86]:
```suggestion
The strategy field is an enum `SubscriptionOpeningStrategy`, having the following values:  
```

File: Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+{PANEL/}
+
+{PANEL: Worker interplay}
+There can only be one active subscription worker working on a subscription. 
+Nevertheless, there are scenarios where it is required to interact between an existing subscription worker and one that tries to connect. 
+This relationship and interoperation is configured by the `SubscriptionConnectionOptions` `Strategy` field.  
+The strategy field is an enum, having the following values:  
+
+* `OpenIfFree` - the server will allow the worker to connect only if there isn't any other currently connected workers.  
+  If there is a existing connection, the incoming worker will throw a SubscriptionInUseException.
```
Comment (garayx) [inline Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown:89]:
```suggestion
  If there is a existing connection, the incoming worker will throw a `SubscriptionInUseException`.
```

File: Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+This relationship and interoperation is configured by the `SubscriptionConnectionOptions` `Strategy` field.  
+The strategy field is an enum, having the following values:  
+
+* `OpenIfFree` - the server will allow the worker to connect only if there isn't any other currently connected workers.  
+  If there is a existing connection, the incoming worker will throw a SubscriptionInUseException.  
+* `WaitForFree` - If the client currently cannot open the subscription because it is used by another client, it will wait for the previous client to disconnect and only then will connect.  
+  This is useful in client failover scenarios where there is one active client and another one already waiting to take its place.  
+* `TakeOver` - the server will allow an incoming connection to overthrow an existing one. It will behave according to the existing connection strategy:
+  * The existing connection has a strategy that is not `TakeOver`. In this case, the incoming connection will take over it causing the existing connection to throw a SubscriptionInUseException exception.
```
Comment (garayx) [inline Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown:93]:
```suggestion
  * The existing connection has a strategy that is not `TakeOver`. In this case, the incoming connection will take over it causing the existing connection to throw a `SubscriptionInUseException`.  
```

File: Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown (hunk 1 above)
Comment (morhilai) [reply Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown:15]: Fixed

File: Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown (hunk 2 above)
Comment (morhilai) [reply Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown:103]: Done

File: Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown (hunk 3 above)
Comment (morhilai) [reply Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown:73]: Done

File: Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown (hunk 4 above)
Comment (morhilai) [reply Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown:85]: Fixed

File: Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown (hunk 5 above)
Comment (morhilai) [reply Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown:86]: Done

File: Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown (hunk 6 above)
Comment (morhilai) [reply Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown:89]: Done

File: Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown (hunk 7 above)
Comment (morhilai) [reply Documentation/5.1/Raven.Documentation.Pages/client-api/data-subscriptions/consumption/how-to-consume-data-subscription.dotnet.markdown:93]: Done

**Review by garayx — APPROVED** [review]

**Pull request merged**